packages/jest-editor-support/src/parsers/babylon_parser.js

```diff
@@ -7,28 +7,68 @@
  * @flow
  */
 
-import {readFileSync} from 'fs';
-
-import {parse as babylonParse} from 'babylon';
+import fs from 'fs';
+import {parse as babelParse} from '@babel/parser';
 import {Expect, ItBlock} from './parser_nodes';
+// TODO: Change import to @babel/parser once types exist
 import type {File as BabylonFile} from 'babylon';
 
 export type BabylonParserResult = {
   expects: Array<Expect>,
   itBlocks: Array<ItBlock>,
 };
 
-export const getASTfor = (file: string): BabylonFile => {
-  const data = readFileSync(file).toString();
-  const config = {plugins: ['*'], sourceType: 'module'};
-  return babylonParse(data, config);
+export const getASTfor = (
+  file: string,
+  enableTypeScript: boolean = false,
+): ?BabylonFile => {
+  const data = fs.readFileSync(file).toString();
+  const config = {
+    // Enable all (most of) the things
+    // https://babeljs.io/docs/en/babel-parser#docsNav
+    plugins: [
+      'asyncGenerators',
+      'bigInt',
+      'classProperties',
+      'classPrivateProperties',
+      'classPrivateMethods',
+      // 'decorators',
+      'doExpressions',
+      'dynamicImport',
+      'exportDefaultFrom',
+      'exportNamespaceFrom',
+      'functionBind',
+      'functionSent',
+      'importMeta',
+      'jsx',
+      'logicalAssignment',
+      'nullishCoalescingOperator',
+      'numericSeparator',
+      'objectRestSpread',
+      'optionalCatchBinding',
+      'optionalChaining',
+      // 'pipelineOperator',
```

Comment:
I am ok leaving `pipelineOperator` out since it is not in babel-6 either. But `decorators` was supported in babel-6, maybe we should add `decorators-legacy` instead?

Reply:
@connectdotz what about adding them from your side instead of hardcoding them here?

Reply:
Do we actually need this list? Since it's just about getting the ast, we could use babel to find the config for us and infer Babylon plug-ins. We've done something similar here recently https://github.com/callstack/linaria/pull/261
Of course I may miss something as I didn't follow the whole PR just yet.

Reply:
@thymikee yes we had an extensive discussion about this, see [here](https://github.com/facebook/jest/pull/6949#discussion_r218641639). In the nutshell, this package is mainly used by standalone jest tools, such as vscode-jest, to parse test file. Just like a javascript project could be using a tool X written in C,  X certainly can't assume all its users are all using C...

> Just because we use babel parser to parse user's file doesn't necessarily mean the user is using babel, right? They might use a different mechanism to transpile their code; in such case, a parser relying on user's .babelrc will not work,

Reply:
> what about adding them from your side instead of hardcoding them here?

@SimenB that is certainly possible, actually I am working on a parser PR for other issue, I plan to submit it to master shortly, and when this PR is ready to go, I will do a separate merge and address some of the related issue then.